src/common/components/contactInfo/contactInfo.component.js

```diff
@@ -99,11 +100,20 @@ class Step1Controller {
           }
         }
 
-        const firstTimeLoading = !find(this.donorDetails.links, ['rel', 'donormatchesform'])
-        if (overrideDonorDetails && firstTimeLoading) {
-          this.donorDetails = assign(this.donorDetails, pick(overrideDonorDetails, [
+        if (overrideDonorDetails) {
+          const afterInitialLoad = !!this.$window.sessionStorage.getItem('afterInitialLoad')
+          const predefinedDonorDetails = pick(overrideDonorDetails, [
             'donor-type', 'name', 'organization-name', 'phone-number', 'spouse-name', 'mailingAddress', 'email'
-          ]))
+          ])
+          if (afterInitialLoad) {
+            this.donorDetails = mergeWith(this.donorDetails, predefinedDonorDetails, (objValue, srcValue) => {
+              if (typeof objValue === 'object') return
+              return objValue || srcValue
+            })
```

Comment:
I think the danger of this block of code is if the overrides have a value and the user tries to override it by deleting that value (e.g. in the case of a bad phone number or middle initial, which are not required). This would make that scenario impossible, right?